lms/djangoapps/courseware/tests/test_module_render.py

```diff
@@ -684,6 +684,330 @@ def test_toc_toy_from_section(self, default_ms, setup_finds, setup_sends, toc_fi
+            True,
+            'error',
+            'There was a problem with your practice proctoring session'
+        ),
+        (
+            'verified',
+            False,
+            None,
+            'Would you like to take "Toy Videos" as a proctored exam?'
+        ),
+        (
+            'verified',
+            False,
+            'submitted',
+            'You have submitted this proctored exam for review'
+        ),
+        (
+            'verified',
+            False,
+            'verified',
+            'Your proctoring session was reviewed and passed all requirements'
+        ),
+        (
+            'verified',
+            False,
+            'rejected',
+            'Your proctoring session was reviewed and did not pass requirements'
+        ),
+        (
+            'verified',
+            False,
+            'error',
+            'There was a problem with your proctoring session'
+        ),
+    )
+    @ddt.unpack
+    def test_render_proctored_exam(self, enrollment_mode, is_practice_exam,
+                                   attempt_status, expected):
+        """
+        Verifies gated content from the student view rendering of a sequence
+        this is labeled as a proctored exam
+        """
+        with self.store.default_store(ModuleStoreEnum.Type.mongo):
+            usage_key = self.course_key.make_usage_key('videosequence', 'Toy_Videos')
+            sequence = self.modulestore.get_item(usage_key)
+
+            sequence.is_time_limited = True
+            sequence.is_proctored_enabled = True
+            sequence.is_practice_exam = is_practice_exam
+
+            self.modulestore.update_item(sequence, self.user.id)
+
+            self.toy_course = self.modulestore.get_course(self.course_key)
+
+            # refresh cache after update
+            self.field_data_cache = FieldDataCache.cache_for_descriptor_descendents(
+                self.course_key, self.request.user, self.toy_course, depth=2
+            )
+
+            from edx_proctoring.api import (
+                create_exam,
+                create_exam_attempt,
+                update_attempt_status
+            )
+            from edx_proctoring.runtime import set_runtime_service
+            from edx_proctoring.tests.test_services import MockCreditService
+
+            set_runtime_service(
+                'credit',
+                MockCreditService(enrollment_mode=enrollment_mode)
+            )
+
+            exam_id = create_exam(
+                course_id=unicode(self.course_key),
+                content_id=unicode(sequence.location),
+                exam_name='foo',
+                time_limit_mins=10,
+                is_proctored=True,
+                is_practice_exam=is_practice_exam
+            )
+
+            if attempt_status:
+                create_exam_attempt(exam_id, self.request.user.id, taking_as_proctored=True)
+                update_attempt_status(exam_id, self.request.user.id, attempt_status)
```

Comment:
Is all the code up to this point identical with the last test point? If so, how about a helper method?